### DISCUSSION TITLE
Rest client Inject Mock in unit test - RequestScoped

### DISCUSSION THREAD

**Discussion by RF-Flores** [quarkusio/quarkus#45006] [Q&A] [answered]
Hi,

I am trying to setup a unit test for a service which has dependency on a certain RequestScoped REST client, is it possible to use InjectMock to setup the REST client mock?

From what i have seen in this blogpost https://quarkus.io/blog/mocking/ and in documentation https://quarkus.io/guides/rest-client#mocking-with-injectmock this is not possible expect when using the Registered Client default scope or it being ApplicationScoped.

[a comment by an automated account is not shown]

**Reply by geoand** [reply]
Does the use case described in the docs not suit your needs? If not, how so?

**Reply by RF-Flores** [reply]
There will be many requests sent concurrently, I don't wish to have to guarantee thread safety using ApplicationScopped, I would rather leverage RequestScope to do that for me.

I am able to get the test setup and fully working with the QuarkusMock setup presented in https://quarkus.io/guides/rest-client#mocking-with-quarkusmock. This is simply a question if I am missing something in my setup or if it is outright not supported.

**Reply by geoand** [reply, marked as answer]
> There will be many requests sent concurrently, I don't wish to have to guarantee thread safety using ApplicationScopped, I would rather leverage RequestScope to do that for me.

Thread-safery is orthogonal to whether the client is request scoped or not.  And in all reality `@RequestScoped` is tied to the current request being served, so it something that needs to be active (for example automatically by Quarkus when an HTTP request is being served or manually by starting the scope) before any such bean is used.
Actually, I would very much recommend **not** using request scoped for the REST Client as you are going to be creating a very large overhead for each request.

**Reply by RF-Flores** [reply]
I see, thank for the clarification. I will rethink my approach accordingly

**Reply by geoand** [reply]
👍🏽
